src/main/scala/scalaz/actors/Actors.scala

```diff
@@ -2,59 +2,45 @@ package scalaz.actors
 
 import scalaz.zio.{ IO, Promise, Queue, Ref, Schedule }
 
-object Actors {
+trait Actor[+E, -F[+ _]] {
+  def ![A](fa: F[A]): IO[E, A]
 
-  trait Actor[+E, -F[+ _]] {
-    def ![A](fa: F[A]): IO[E, A]
-    def stop(): IO[Nothing, Unit]
-  }
-
-  trait MessageHandler[S, +E, -F[+ _]] {
-    def receive[A](state: S, msg: F[A]): IO[E, (S, A)]
-  }
-
-  trait Supervisor[E] {
-    def supervise[A](io: IO[E, A], error: E): IO[E, A]
-  }
-
-  object Supervisor {
-
-    def none: Supervisor[Nothing] = retry(Schedule.never)
+  val stop: IO[Nothing, Unit]
```

Comment:
It could be useful to return a `List` of unprocessed messages, i.e. whatever is in the actor's mailbox.

Reply:
@jdegoes would it be an idea to add to zio `Queue` a method to provide the `List` of items in it and then use this here?

Reply:
@jdegoes thanks for the tip on `takeAll` ... so this is what I have to do that and do a shutdown:

```                                                                                                                                 
override def stop[PendingMessage]: IO[Nothing, List[PendingMessage]] =                                                                                                 
  for {                                                                                                                                                                                   
     tall <- queue.takeAll                                                                                                                                              
      _ <- queue.shutdown                                                                                                                                                
} yield tall     
```
I get this mismatch when compiling
```
bt:scalaz-actors> compile                                                                                                                                                       
 269 [info] Compiling 1 Scala source to /home/salar/projects/scalaz-actors/target/scala-2.1                                                                                           
 270 2/classes ...                                                                                                                                                                    
 271 [error] /home/salar/projects/scalaz-actors/src/main/scala/scalaz/actors/Actors.scala:6                                                                                           
 272 2:19: type mismatch;                                                                                                                                                             
 273 [error]  found   : List[PendingMessage[E, F, _]]                                                                                                                                 
 274 [error]     (which expands to)  List[(F[_$1], scalaz.zio.Promise[E,_$1])( forSome { ty                                                                                           
 275 pe _$1 })]                                                                                                                                                                       
 276 [error]  required: List[PendingMessage]                                                                                                                                          
 277 [error]           } yield tall                                                                                                                                                   
 278 [error]                   ^                                                                                                                                                      
 279 [info] List[PendingMessage[E, F, _]] <: List[PendingMessage]?    
```
Whats the best way to handle this? Thanks!

Reply:
Change `List[PendingMessage]` to `List[PendingMessage[E, F, _]]`.

Reply:
@jdegoes I tried that and got:

```
PendingMessage does not take type parameters
```

Reply:
Well `scalac` is lying because I see:

```scala
type PendingMessage[E, F[_], A] = (F[A], Promise[E, A])
```

Ah, I see the problem!

You have a type parameter called `PendingMessage`, which is shadowing the `PendingMessage` inside the companion object. Just delete that type parameter and it should work. 

😅

Reply:
@jdegoes I am blind 😃 is it the type parameter where I override `stop` ? If so if I remove that it won't match my trait definition or know what type the list being returned?

Reply:
@jdegoes oh ignore my last message I think I got it. Sorry I was an idiot/blind. More coffee :-)